### PR TITLE
:book: Fixes for clusterctl developer doc, mostly related to docker provider

### PR DESCRIPTION
**Pull request by dlipovetsky**
**What this PR does / why we need it**:
Improves the accuracy and completeness of the docs for developers using clusterctl and the docker provider.

### CODE REVIEW

**Comment by dlipovetsky** [comment]
Force-pushing some fixes before anyone has started reviewing :speak_no_evil:

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ export KUBECONFIG=<...>
 
 [kind] is not designed for production use.
 
-**Minimum [kind] supported version**: v0.6.x
+**Minimum [kind] supported version**: v0.6.1
```
Comment (vincepri) [inline docs/book/src/user/quick-start.md:39]: I think we're now suggesting v0.7.+ ?

**Comment by vincepri** [comment]
One note, changes lgtm — only need to squash and we should be ready to merge

File: docs/book/src/user/quick-start.md (hunk 1 above)
Comment (dlipovetsky) [reply docs/book/src/user/quick-start.md:39]: Ok. I was going by the version used by CI scripts:
```
hack/ensure-kind.sh
24:MINIMUM_KIND_VERSION=v0.6.1
36:      curl -sLo "${GOPATH_BIN}/kind" https://github.com/kubernetes-sigs/kind/releases/download/${MINIMUM_KIND_VERSION}/kind-linux-amd64
46:  if ! [[ "${kind_version}" =~ ${MINIMUM_KIND_VERSION} ]]; then
49:Requires ${MINIMUM_KIND_VERSION} or greater.
50:Please install ${MINIMUM_KIND_VERSION} or later.

scripts/ci-integration.sh
26:KIND_VERSION="v0.6.1"
41:   wget "https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-${GOOS}-${GOARCH}" \
```
Do these scripts need to be updated to 0.7.0? (I can open a separate PR)

Comment (vincepri) [reply docs/book/src/user/quick-start.md:39]: Looking at the code, right now we're using 0.7.1, maybe let's match that?

Comment (vincepri) [reply docs/book/src/user/quick-start.md:39]: (CAPD uses kind as a library I meant)

Comment (dlipovetsky) [reply docs/book/src/user/quick-start.md]: Changed to 0.7.0.

**Comment by vincepri** [comment]
/approve

LGTM pending squash

**Comment by dlipovetsky** [comment]
Squashed.

**Comment by vincepri** [comment]
Thank you @dlipovetsky !

**Pull request merged**